spec/controllers/project_controller_spec.rb

```diff
@@ -1,4 +1,116 @@
 require 'rails_helper'
-RSpec.describe ProjectController, type: :controller do
-  
+RSpec.describe ProjectsController, type: :controller do
+  describe 'For Project' do
+    before(:each) do
+      @company = Company.create(FactoryGirl.attributes_for(:company))
+      user_params = FactoryGirl.attributes_for(:user)
+      user_params[:company_id] = @company.id
+      @user = User.create(user_params)
+      @request.env['devise.mapping'] = Devise.mappings[:user]
+      @request.host = "#{@company.subdomain}.lvh.me:3000"
+      @user.confirm
+      sign_in @user
+      project_params = FactoryGirl.attributes_for(:project)
+      project_params[:company] = @company
+      @project = Project.create(project_params)
```

Comment:
This instance of project is not being used in all test cases, so move it to the relevant sub section of test cases for which it is actually required. Same for @team.